
<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${severity}][${thread}][${node}/${function}:${line}]: ${message}"/>
  <env name="ROS_LOG_DIR" value="/tmp/"/>

  <arg name="header_frame_id" default="wheel_controller" />

  <arg name="gps_state_input_topic" default="/robot_platform/gps"/>
  <arg name="imu_state_input_topic" default="/robot_platform/imu"/>
  <arg name="wheel_positions_input_topic" default="/robot_platform/wheel_response"/>
  <arg name="odom_state_output_topic" default="/robot_platform/odom"/>

  <node pkg="robot_platform" type="odometry_handler" name="odometry_handler" output="screen" respawn="true">

    <param name="header_frame_id" type="string" value="$(arg header_frame_id)" />
    
    <param name="gps_state_input_topic" type="string" value="$(arg gps_state_input_topic)" />
    <param name="imu_state_input_topic" type="string" value="$(arg imu_state_input_topic)" />
    <param name="wheel_positions_input_topic" type="string" value="$(arg wheel_positions_input_topic)" />

    <param name="odom_state_output_topic" type="string" value="$(arg odom_state_output_topic)" />

    
  </node>

</launch>
